table/block_based/block_based_table_builder.cc

```diff
@@ -1708,6 +1701,8 @@ void BlockBasedTableBuilder::Abandon() {
     rep_->pc_rep->finished = true;
   }
   rep_->state = Rep::State::kClosed;
+  rep_->GetStatus().PermitUncheckedError();
+  rep_->GetIOStatus().PermitUncheckedError();
```

Comment:
This would be better to do in a method off of rep to avoid two lock calls.

Reply:
> This would be better to do in a method off of rep to avoid two lock calls.

Sure it sounds good, but might not be directly related to this PR and can be done separately.

Reply:
> This would be better to do in a method off of rep to avoid two lock calls.

Sounds a good idea, but I am not quite concerned about it in Abandon() path. Abandon() is only called in error cases, so the performance is not so critical here.